blocks/rich-text/index.js

```diff
@@ -558,6 +559,44 @@ export class RichText extends Component {
 		if ( keyCode === BACKSPACE ) {
 			this.onChange();
 		}
+
+		// `scrollToCaret` is called on `nodechange`, whereas calling it on
+		// `keyup` *when* moving to a new RichText element results in incorrect
+		// scrolling. Though the following allows false positives, it results
+		// in much smoother scrolling.
+		if ( keyCode !== BACKSPACE && keyCode !== ENTER ) {
+			this.scrollToCaret();
+		}
+	}
+
+	scrollToCaret() {
+		if ( ! this.props.isViewportSmall ) {
+			return;
+		}
+
+		const { top: caretTop } = this.getEditorSelectionRect();
```

Comment:
I think `getFocusPosition` is now equivalent to `getEditorSelectionRect`? So we could pass that in `nodeChange`?

Reply:
I'm a bit tied up today, but if you're interested and have spare time, I'd be grateful if you had a go. :)

Reply:
Sure!